src/test/test-log.c

```diff
@@ -70,6 +70,25 @@ static void test_log_syntax(void) {
         assert_se(log_syntax("unit", LOG_ERR, "filename", 10, SYNTHETIC_ERRNO(ENOTTY), "ENOTTY: %s: %m", "hogehoge") == -ENOTTY);
 }
 
+static void test_log_context(void) {
+        char *strv[] = { (char*) "MYDATA=abc", NULL };
+
+        {
+                LOG_CONTEXT_PUSH("MYDATA=abc");
+                LOG_CONTEXT_PUSH("MYDATA=def");
+                LOG_CONTEXT_PUSH_STRV(strv);
+                LOG_CONTEXT_PUSH_STRV(strv);
```

Comment:
Should it work if calling a function after this line, and doing more LOG_CONTEXT_PUSH from there? If so, would probably be good to cover that case too in the test

Reply:
Extended the test a bit more in general. Went for a nested block since that's pretty much the same as a function in this scenario.